`Traceback (most recent call last):
  File "<embedded stdlib>/unittest/case.py", line 59, in testPartExecutor
    yield
  File "<embedded stdlib>/unittest/case.py", line 605, in run
    testMethod()
  File "/build/work//google3/runfiles/google3/third_party/tensorflow/python/framework/memory_checker_test.py", line 120, in testNoNewPythonObjectsEmpty
    threshold={'builtins.weakref': 1})
  File "/build/work//google3/runfiles/google3/third_party/tensorflow/python/profiler/traceme.py", line 59, in wrapper
    return func(*args, **kwargs)
  File "/build/work//google3/runfiles/google3/third_party/tensorflow/python/framework/memory_checker.py", line 141, in assert_no_new_python_objects
    self._python_memory_checker.assert_no_new_objects(threshold=threshold)
  File "/build/work//google3/runfiles/google3/third_party/tensorflow/python/profiler/traceme.py", line 59, in wrapper
    return func(*args, **kwargs)
  File "/build/work//google3/runfiles/google3/third_party/tensorflow/python/framework/python_memory_checker.py", line 131, in assert_no_new_objects
    threshold, original_count_diff.most_common()))
AssertionError: New Python objects exceeded the threshold.
Python object threshold:
{'builtins.weakref': 1}

New Python objects:
[('builtins.function', 1)]`

--- a/tensorflow/python/distribute/cluster_resolver/tfconfig_cluster_resolver.py
+++ b/tensorflow/python/distribute/cluster_resolver/tfconfig_cluster_resolver.py
@@ -88,7 +88,7 @@ class TFConfigClusterResolver(ClusterResolver):
 
   @property
   def task_id(self):
-    if self._task_type is None:
+    if self._task_id is None:
       task_info = _get_value_in_tfconfig(_TASK_KEY, {})
       return int(task_info['index']) if 'index' in task_info else None
     else:

--- a/tensorflow/python/distribute/cluster_resolver/tfconfig_cluster_resolver_test.py
+++ b/tensorflow/python/distribute/cluster_resolver/tfconfig_cluster_resolver_test.py
@@ -224,6 +224,21 @@ class TFConfigClusterResolverTest(test.TestCase):
     cluster_resolver = TFConfigClusterResolver()
     self.assertEqual(1, cluster_resolver.task_id)
 
+  def testTaskIndexOverride(self):
+    os.environ['TF_CONFIG'] = """
+    {
+      "cluster": {
+        "worker": ["worker0:2222", "worker1:2222"]
+      },
+      "task": {
+        "type": "worker",
+        "index": "0"
+      }
+    }
+    """
+    cluster_resolver = TFConfigClusterResolver(task_id=1)
+    self.assertEqual(1, cluster_resolver.task_id)
+
   def testZeroItemsInClusterSpecMasterRead(self):
     os.environ['TF_CONFIG'] = """
     {}

